core/src/json_schema.rs

```diff
@@ -157,54 +266,41 @@ pub mod simplified {
                     additional_properties: None,
                     items: None,
                     any_of: None,
-                } => Ok(Atomic::Any),
+                } => Ok(Atomic::Scalar(BsonTypeName::Any)),
                 json_schema::Schema {
                     bson_type: Some(bson_type),
                     properties,
                     required,
                     additional_properties,
                     items,
                     any_of: None,
-                } => {
-                    let construct_obj = || {
```

Comment:
this `construct_obj` closure made sense before I moved the `Multiple` code out of here. Now it isn't necessary :D